profiling/src/collections/identifiable/mod.rs

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+mod string_id;
```

Comment:
Why is this the only ID that is under here?

Reply:
I picked it out of a +3,000 line change branch and didn't notice ^_^

The reason it's there in that branch is because `StringTable` is a data-structure which uses `StringId`, so it made sense to move it. But for the generic `Table` version, it works abstractly on `crate::collections::{Id, Item}`, so it doesn't need to know anything about `Function` or `FunctionId`, etc.